commands/http/handler.go

```diff
@@ -84,6 +80,27 @@ func (i Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte(err.Error()))
 		return
 	}
+
+	// get the node's context to pass into the commands.
+	node, err := i.ctx.GetNode()
+	if err != nil {
+		err = fmt.Errorf("cmds/http: couldn't GetNode(): %s", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	ctx, cancel := context.WithCancel(node.Context())
+	defer cancel()
+	/*
+		TODO(cryptix): the next line looks very fishy to me..
+		It looks like the the context for the command request beeing prepared here is shared across all incoming requests..
+
+		I assume it really isn't because ServeHTTP() doesn't take a pointer receiver, but it's really subtule..
+
+		Shouldn't the context be just put on the command request?
```

Comment:
Yeah, this is super sketchy. I think its probably best to put the context into the request object instead of here on the handler context